docs/contributing.md

```diff
@@ -354,17 +354,38 @@ If you're using IntelliJ, you can import [these code style settings](https://git
-* You can create a release branch via the GitHub UI.
-* From this branch a git tag is created for the specific release, for example `v0.3.0`.
-* Tagging a release will automatically build and push the relevant artifacts to their repositories or package managers \(docker images, Python wheels, etc\).
-* A release branch should be substantially _feature complete_ with respect to the intended release. Code that is committed to `master` may be merged or cherry-picked on to a release branch, but code that is directly committed to the release branch should be solely applicable to that release \(and should not be committed back to master\).
-* In general, unless you're committing code that only applies to the release stream \(for example, temporary hotfixes, backported security fixes, or image hashes\), you should commit to `master` and then merge or cherry-pick to the release branch.
-* It is also important to update the [CHANGELOG.md](https://github.com/gojek/feast/blob/master/CHANGELOG.md) when submitting a new release. This can be in the same PR or a separate PR.
-* Finally it is also important to create a [GitHub release](https://github.com/gojek/feast/releases) which includes a summary of important changes as well as any artifacts associated with that release.
+* Each major and minor release has a long-lived maintenance branch, for example `v0.3-branch`. This is called a "release branch".
+* From the release branches, patch version releases are tagged, for example `v0.3.0`.
+
+A release branch should be substantially _feature complete_ with respect to the intended release. Code that is committed to `master` may be merged or cherry-picked on to a release branch, but code that is directly committed to a release branch should be solely applicable to that release \(and should not be committed back to master\).
+
+In general, unless you're committing code that only applies to a particular release stream \(for example, temporary hotfixes, backported security fixes, or image hashes\), you should base changes from `master` and then merge or cherry-pick to the release branch.
+
+### Release process
+
+For Feast maintainers, these are the concrete steps for making a new release.
+
+1. For a major or minor release, create and check out the release branch for the new stream, e.g. `v0.6-branch`. For a patch version, check out the stream's release branch.
+1. In the root `pom.xml`, remove `-SNAPSHOT` from the `<revision>` property, and commit.
+1. Push. For a new release branch, open a PR against master.
+1. When CI passes, merge. (Remember _not_ to delete the new release branch).
+1. Tag the merge commit with the release version, using a `v` prefix. Push the tag.
+1. Bump to the next working version and append `-SNAPSHOT` in `pom.xml`.
+1. Update the [CHANGELOG.md]:
+
+       $ docker run -it --rm -e CHANGELOG_GITHUB_TOKEN="[Token]" -v "$(pwd)":/usr/local/src/your-app ferrarimarco/github-changelog-generator --user gojek --project feast
```

Comment:
Perhaps we can set options to map Feast's label conventions to the generator's grouping ones, like `--enhancement-labels kind/feature --bug-labels kind/bug`

Reply:
That would be super cool. 

I am just a bit worried that we havent been tagging everything diligently yet. Perhaps we should add this a bit later.